### ISSUE TITLE
Backend needs range size cast support for prange arg

### ISSUE THREAD

**Issue by stuartarchibald** [numba/numba#2409]
As noted in https://github.com/numba/numba/pull/2391#issuecomment-307942916

This:
```python
from numba import njit, prange  
import numpy as np

@njit(parallel=True)
def test_impl(n):
    acc = 0
    for i in prange(n):
        acc += 1
    return acc

test_impl(np.int32(4))
```
produces:
```
  File "<path>/numba/lowering.py", line 214, in lower_block
    self.lower_inst(inst)
  File "<path>/numba/lowering.py", line 358, in lower_inst
    parfor.lower_parfor_parallel(self, inst)
  File "<path>/numba/npyufunc/parfor.py", line 79, in _lower_parfor_parallel
    call_parallel_gufunc(lowerer, func, gu_signature, func_sig, func_args, loop_ranges, parfor_redvars, parfor_reddict, parfor.init_block)
  File "<path>/numba/npyufunc/parfor.py", line 428, in call_parallel_gufunc
    stop = builder.sub(stop, one)
  File "<path>/site-packages/llvmlite/ir/builder.py", line 24, in wrapped
    % (lhs.type, rhs.type))
numba.errors.LoweringError: Failed at nopython (nopython mode backend)
Operands must be the same type, got (i32, i64)
File "foo2.py", line 7
[1] During: lowering "[LoopNest(index_variable=parfor_index.0, range=0,n,1 correlation=-1)]{16: <ir.Block at foo2.py (7)>}Var(parfor_index.0, foo2.py (7))" at foo2.py (7)
```
the backend needs to upcast the `prange` range size to `int64`.

**Comment by DrTodd13** [comment]
I believe this is fixed in d763c3703b3648ecfcba4b848362935a04bcd04f.

**Comment by stuartarchibald** [comment]
Thanks @DrTodd13 I agree. Closing.
